lib/ansible/plugins/lookup/first_found.py

```diff
@@ -170,16 +170,16 @@ def _process_terms(self, terms, variables, kwargs):
         for term in terms:
             if isinstance(term, Mapping):
                 self.set_options(var_options=variables, direct=term)
+                files = self.get_option('files')
             elif isinstance(term, string_types):
-                self.set_options(var_options=variables, direct=kwargs)
+                files = [term]
             elif isinstance(term, Sequence):
                 partial, skip = self._process_terms(term, variables, kwargs)
                 total_search.extend(partial)
                 continue
             else:
                 raise AnsibleLookupError("Invalid term supplied, can handle string, mapping or list of strings but got: %s for %s" % (type(term), term))
 
-            files = self.get_option('files')
```

Comment:
There's a TODO below which fixes some of the test breakage:

```python
                # NOTE: this seems wrong, should be 'extend' as any option/entry can clobber all
                total_search = filelist
```

This fixes it
```diff
-                total_search = filelist
+                total_search.extend(filelist)
```